### PR TITLE
refactor: extent and test match Controller

### PR DESCRIPTION
**Pull request by chmette**
- added missing controller functions 
- tested in insomnia
if you have ideas to optimize or shorten code please always let me know 
maybe TODO in separate branch : clean up validation of create branch in middleware

### CODE REVIEW

File: controllers/matchControllers.js [hunk 1]
```diff
@@ -1,5 +1,9 @@
 const Match = require('../models/Match');
+const User = require('../models/User');
+const customError = require('../helpers/customErrorHandler');
+const mongoose = require('mongoose');
 
+// just for checking reason
 exports.getMatches = async (req, res, next) => {
```
Comment (luke-dowling) [inline controllers/matchControllers.js:7]: I think this it two calls in one, one is get matches and one is get match singular. also i dont think two res.jsons will work in one call

File: controllers/matchControllers.js [hunk 2]
```diff
@@ -9,22 +13,154 @@ exports.getMatch = async (req, res, next) => {
...
+        next(customError(`Books already matched`, 400));
+        return;
+      } else {
+        createMatch(req.body);
+      }
+    }
+  } catch (err) {
+    if (err instanceof mongoose.Error.CastError) {
+      next(customError(`ID: ${id} is not valid`, 400));
+      next(err);
```
Comment (luke-dowling) [inline controllers/matchControllers.js]: this need to be outside the if block

File: controllers/matchControllers.js [hunk 3]
```diff
@@ -9,22 +13,154 @@ exports.getMatch = async (req, res, next) => {
...
+};
+
+exports.updateMatch = async (req, res, next) => {
+  const { id } = req.params;
+
+  try {
+    const match = await Match.findById(id);
+
+    if (!match) {
+      next(customError(`Match with ID: ${id} does not exist`, 400));
```
Comment (luke-dowling) [inline controllers/matchControllers.js:123]: i think you need a return here

File: controllers/matchControllers.js [hunk 4]
```diff
@@ -9,22 +13,154 @@ exports.getMatch = async (req, res, next) => {
...
+      return;
+    }
+
+    await User.findByIdAndUpdate(match.bookOne.owner, {
+      $pull: { matches: match._id },
+    });
+    await User.findByIdAndUpdate(match.bookTwo.owner, {
+      $pull: { matches: match._id },
+    });
+    match.delete();
```
Comment (luke-dowling) [inline controllers/matchControllers.js]: await match.delete() ?

File: routes/booksRouter.js [hunk 5]
```diff
@@ -5,13 +5,12 @@ const {
...
   getUserLibrary,
   getBooks,
-  addBook,
   deleteBook,
   updateBook,
   addInterestedUser,
 } = require('../controllers/bookControllers');
 
-router.route('/').get(getBooks).post(addBook);
+router.route('/').get(getBooks);
```
Comment (luke-dowling) [inline routes/booksRouter.js]: why did you delete add book?

**Review by luke-dowling — COMMENTED** [review]
just some things to check but great controller!

File: controllers/matchControllers.js (hunk 2 above)
Comment (chmette) [reply controllers/matchControllers.js]: done, thanks

File: controllers/matchControllers.js (hunk 3 above)
Comment (chmette) [reply controllers/matchControllers.js:123]: true, done

File: controllers/matchControllers.js (hunk 4 above)
Comment (chmette) [reply controllers/matchControllers.js]: true, works without as well, but you are completely right

File: routes/booksRouter.js (hunk 5 above)
Comment (chmette) [reply routes/booksRouter.js]: hupsiiiiii

**Review by luke-dowling — APPROVED** [review]
niceeeeeeeeee
